backend/backend/main.py

```diff
@@ -33,3 +34,6 @@ async def lifespan(app: FastAPI):
 
 # Include API routes
 app.include_router(api_router)
+
+# Mount Gradio routes
+app = gr.mount_gradio_app(app, gradio_routes, path="/gradio")
```

Comment:
It looks like you're mounting Gradio routes twice. Consider removing the redundant assignment.